### ISSUE TITLE
Inter occasion (or intra subject) variability 

### ISSUE THREAD

**Issue by nmens** [Open-Systems-Pharmacology/Forum#180]
Dear all
I've been testing PKSim for the last months and becoming a fan!
For simulating clinical trials, however, it would be interesting if inter-occasion (or intra-subject) variability could be set for a certain drug/population.
Any clue how this is possible? Simulating another population with the same characteristics (BMI range) is not the same as set an intra-subject variability because different individuals (with different weights and consequently different volumes of distribution) would constitute the other population. Variability would be inter-subject.
Intra-subject CV is of particular interest for crossover studies.
Thanks for the help in advance
Best regards to all the group forum
NS

**Comment by StephanSchaller** [comment]
I have to add... the "occasio" variability should be added in the simulation once you have created it.

I am sorry to say, you will not be able to add variability to any parameter you like.
E.g. what could be of interest is the dose per BW or Dosing Time, you won't be able to add variability here. This si something you would have to do via R or MATLAB.

**Comment by msevestre** [comment]
>E.g. what could be of interest is the dose per BW or Dosing Time, you won't be able to add variability here. This is something you would have to do via R or MATLAB.

For some parameters, it would make sense to have a chance to vary them in PK-Sim, at least in some kind of advanced mode. I have created an issue [here](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/759) where we should list those parameters that should be available to vary out of the box

You do not have to go to Matlab or R however to accomplish this. Another alternative that may be easier depending on your mileage is:

1. export the simulation to MoBi
1. update the parameters that you want to vary by checking the "CanBeVariedInPopulation" checkbox
1. import the simulation back to PK-Sim. 

Using this pattern, you can pretty much vary anything in PK-Sim. 

Cheers,
Michael

**Comment by nmens** [comment]
I really appreciate the comments of Shaller and Sevestre. Many thanks!

Just to clarify that inter-occasion (or intra-idividual) variability is the variability that is seen in the same individual with the administration of the same formulation in two different occasions.
When creating a population with 16 subjects, for example, and you give to each subject the same tablet in two different occasions, the pharmacokinetic profile might be quite different between these 2 occasions For simplicity, in PBPK modelling, if you add variability to transcellular permeability, to renal clearance and to hepatic clearance, you might be able to simulate this in vivo characteristic.
There are drugs that are considered as highly variable drugs because you have a variability in the response parameters (normally Cmax) characterized by a coefficient of variation of >=30%.
Therefore, following Sevestre advice, and selecting, in Mobi, intestinal Permeability parameter and check "CanBeVariedInPopulation", how do you input the variability (CV value?)
Thanks for the help, in advance!
Cheers
NS

**Comment by StephanSchaller** [comment]
I guess for intestinal permeability, you do not have to go to MoBi. You can vary intestinal absorption either with surface area or permeability: *Surface Area*: In your Population (before creating the simulation) within the *User Defined Variability* Tab, simply either choose each segment separately or choose the "Effective surface area variability factor" for the whole lumen:
![image](https://user-images.githubusercontent.com/25477643/40329901-7cb4b3a6-5d4b-11e8-925d-96c26ea79ac5.png)

Permeability can be defined once the simulation is created with the population:
![image](https://user-images.githubusercontent.com/25477643/40330098-25fd6c0a-5d4c-11e8-97b9-56a93fbf8b1a.png)

**Comment by nmens** [comment]
Thanks for the help!
I've tested and it works for the purpose I want. I've checked also that after simulation you can add variability to specific clearance (renal and hepatic).
Therefore I used the same population generated in the building block in two different simulations. To check if I was starting from exactly the same individuals (but with variability in Peff, CLr and CLh) in the two independent simulations, I exported Population from each simulation in order to compare anatomic and physiological parameter values.

HOWEVER, from the exportation lacks identification of at least 75 columns. The last two columns are the ones that differ between the two exports. How can I know fields names for each column?
Rgds
NS

**Comment by msevestre** [comment]
>HOWEVER, from the exportation lacks identification of at least 75 columns. The last two columns are the ones that differ between the two exports. How can I know fields names for each column?

What do you mean? Export is the a csv file. One column per parameter. Make sure you are opening the file properly in excel using the appropriate file separator (;). 

If this is not clear, please attach the exported file

**Comment by nmens** [comment]
Hi Sevestre
GitHub does not allow to drag and drop .csv files. So I copied already the excel sheet after transforming text to columns using (;) as separator and saving as .xlsx file. 

[Population Tablet 50 mg Reference with variability-Population.xlsx](https://github.com/Open-Systems-Pharmacology/Forum/files/2028508/Population.Tablet.50.mg.Reference.with.variability-Population.xlsx)

Did I do something wrong?

**Comment by msevestre** [comment]
@nmens 
Please zip your csv file. Then upload the corresponding file to github

Also as an FYI my first name is Michael.
Cheers,

**Comment by nmens** [comment]
Hi Michael
Thanks for the help!
Attached is a zip file of the csv, as you suggested.
Rgds
Nuno
[Population Tablet 50 mg Reference with variability-Population.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/2028546/Population.Tablet.50.mg.Reference.with.variability-Population.zip)

**Comment by msevestre** [comment]
CSV File looks ok to me
How to import a csv file in Excel:

1. Create new workbook
2. Goto Data, From Text
![image](https://user-images.githubusercontent.com/1041237/40390574-b6a348dc-5de3-11e8-8135-eef7dcc4e33c.png)
3. Select you file
4. Delimited, next
5. Check semicolon, next
6. Finish

The file will be imported correctly

Also attached here
[Population Tablet 50 mg Reference with variability-Population.xlsx](https://github.com/Open-Systems-Pharmacology/Forum/files/2028604/Population.Tablet.50.mg.Reference.with.variability-Population.xlsx)

**Comment by nmens** [comment]
Michael

Many thanks! I followed your instructions and everything is OK. The last 3 columns belong to the parameters for which I add variability. also thanks for the file!

So, I think that inter-occasion variability can be added to population as this way. Now I just need to find out the magnitude of geometric standard deviation to induce a certain intra-subject CV in a crossover design.
Best regards

**Comment by StephanSchaller** [comment]
Great!

If I am correct, we will see each other in the workshop at page next monday?

Until then! Stephan

**Comment by nmens** [comment]
Yes!
Course program is really challenging. And it will be good to learn with the best!
Unfortunately I was not able to attend PK Sim course in Lisbon in 2016, during PAGE meeting, as I was speaking in other parallel session on the same day. Fortunately, in the meeting, I was able to be with PK Sim personnel and they have shown me the potential of the software. My background was Trial Simulator, in which I have developed an absorption and disposition model and so it was easier to move on into PK Sim. But I feel the need to lean with someone that is able to explore all the capacities of this software.
Until then! Nuno
